tests/unit/test_charm.py

```diff
@@ -1,7 +1,251 @@
+        )
+        self.assertEqual(
+            state_out.deferred[0].name,
+            "nssf_pebble_ready",
+        )
+
+    @patch("charm.check_output")
+    def test_given_relations_created_and_nrf_data_available_when_pebble_ready_then_config_file_rendered_and_pushed(  # noqa: E501
+        self,
+        patch_check_output,
+    ):
+        config_dir = tempfile.TemporaryDirectory()
+        container = Container(
+            name="nssf",
+            can_connect=True,
+            mounts={"config_dir": Mount("/free5gc/config", config_dir.name)},
+        )
+        nrf_relation = Relation(
+            endpoint="fiveg_nrf",
+            remote_app_name="remote",
+            remote_app_data={"url": "http://nrf:8081"},
+        )
+        state_in = State(
+            containers=[container],
+            relations=[nrf_relation],
+            model=Model(name="whatever"),
+        )
+        patch_check_output.return_value = "1.1.1.1".encode()
+
+        self.ctx.run(container.pebble_ready_event, state_in)
+
+        with (
+            open(Path(config_dir.name) / "nssfcfg.conf") as actual,
+            open(Path(__file__).parent / "expected_config" / "config.conf") as expected,
+        ):
+            actual_content = actual.read()
+            expected_content = expected.read().strip()
+            self.assertEqual(actual_content, expected_content)
+
+    @patch("charm.check_output")
+    def test_config_pushed_but_content_changed_when_pebble_ready_then_new_config_content_is_pushed(  # noqa: E501
+        self,
+        patch_check_output,
+    ):
+        config_dir = tempfile.TemporaryDirectory()
+        container = Container(
+            name="nssf",
+            can_connect=True,
+            mounts={"config_dir": Mount("/free5gc/config", config_dir.name)},
+        )
+        nrf_relation = Relation(
+            endpoint="fiveg_nrf",
+            remote_app_name="remote",
+            remote_app_data={"url": "http://nrf:8081"},
+        )
+        state_in = State(
+            containers=[container],
+            relations=[nrf_relation],
+            model=Model(name="whatever"),
+        )
+        patch_check_output.return_value = "1.1.1.1".encode()
+        with open(Path(config_dir.name) / "nssfcfg.conf", "w") as existing_config:
+            existing_config.write("never gonna give you up")
+
+        self.ctx.run(container.pebble_ready_event, state_in)
+
+        with (
+            open(Path(config_dir.name) / "nssfcfg.conf") as actual,
+            open(Path(__file__).parent / "expected_config" / "config.conf") as expected,
+        ):
+            actual_content = actual.read()
+            expected_content = expected.read().strip()
+            self.assertEqual(actual_content, expected_content)
+
+    @patch("charm.check_output")
+    def test_given_relation_available_and_config_pushed_when_pebble_ready_then_pebble_layer_is_added_correctly(  # noqa: E501
```

Comment:
Are we testing the pebble layer is added or whether if it's the expected one or not?

Reply:
We are testing that the correct layer is applied.